src/components/Article/DiscussionModal.tsx

```diff
@@ -0,0 +1,227 @@
+  const [newComment, setNewComment] = useState("");
+  const [user, setUser] = useState<User | null>(null);
+
+  const placeholderAvatar = "https://via.placeholder.com/150"; // Placeholder avatar URL
+
+  useEffect(() => {
+    const fetchUserData = async () => {
+      const userUid = localStorage.getItem("userUid");
+      if (userUid) {
+        const db = getDatabase();
+        const userRef = ref(db, `users/${userUid}`);
+        try {
+          const userSnapshot = await get(userRef);
+          if (userSnapshot.exists()) {
+            const userData = userSnapshot.val();
+            setUser({
+              name: userData.name || "Guest User",
+              pic: userData.pic || placeholderAvatar,
+            });
+          }
+        } catch (error) {
+          console.error("Error fetching user data:", error);
+        }
+      }
+    };
+
+    fetchUserData();
+
+    const fetchComments = async () => {
+      const db = getDatabase();
+      const commentsRef = ref(db, `articles/${blogId}/comments`);
+      onValue(commentsRef, async (snapshot) => {
+        const commentIdsString = snapshot.val();
+        if (commentIdsString) {
+          const commentIds = commentIdsString.split(",");
+          const commentsArray: CommentData[] = [];
+          for (const commentId of commentIds) {
+            const commentRef = ref(db, `comments/${commentId}`);
+            const commentSnapshot = await get(commentRef);
+            if (commentSnapshot.exists()) {
+              commentsArray.push({
+                id: commentId,
+                ...commentSnapshot.val(),
+              });
+            }
+          }
+          setComments(commentsArray);
+        } else {
+          setComments([]);
+        }
+      });
+    };
+
+    fetchComments();
+  }, [blogId]);
+
+  function generateUUID() {
+    var d = new Date().getTime();
+    var d2 =
+      (typeof performance !== "undefined" &&
+        performance.now &&
+        performance.now() * 1000) ||
+      0;
+    return "xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx".replace(
+      /[xy]/g,
+      function (c) {
+        var r = Math.random() * 16;
+        if (d > 0) {
+          r = (d + r) % 16 | 0;
+          d = Math.floor(d / 16);
+        } else {
+          r = (d2 + r) % 16 | 0;
+          d2 = Math.floor(d2 / 16);
+        }
+        return (c === "x" ? r : (r & 0x3) | 0x8).toString(16);
+      }
+    );
+  }
+
+  const handleSubmit = async (e: FormEvent) => {
+    e.preventDefault();
+
+    if (newComment.trim() && user) {
+      const db = getDatabase();
+      const newCommentId = generateUUID();
+      const newCommentData: CommentData = {
+        id: newCommentId,
+        author: user.name,
+        avatar: user.pic,
+        content: newComment,
+        timestamp: new Date().toISOString(),
+      };
+
+      // Store comment details in `comments` node
+      const commentRef = ref(db, `comments/${newCommentId}`);
```

Comment:
@nishant0708 The comments node must be inside the single article and it should only contain the comments for that article only.

* articles
  * {an_article}
    * {content...}
    * comments